UI/WebServerResources/SOGoMailDataSource.js

```diff
@@ -82,9 +82,35 @@ SOGoMailDataSource = Class.create({
         },
 
         _loadCallback: function(http) {
+            var data,
+            re, reMatches, formLine;
             if (http.status == 200) {
                 if (http.responseText.length > 0) {
-                    var data = http.responseText.evalJSON(true);
+                    try {
+                        data = http.responseText.evalJSON(true);
+                    } catch(e) {
+                        if (e instanceof SyntaxError) {
+                            re = /<form (.*id=["']connectForm["'].*)>/i;
```

Comment:
All this logic would be easy to move it else where, into a function, so it can be reused easily.